Makefile

```diff
@@ -109,6 +109,9 @@ test-docker-py: build ## run the docker-py tests
 test-integration-cli: build ## run the integration tests
 	$(DOCKER_RUN_DOCKER) hack/make.sh dynbinary test-integration-cli
 
+test-splunk-logger: build ## run the logger tests
```

Comment:
I think all the log driver code is already in the codebase, the additions for this PR are simply test files and scripts to enable developers to run some tests against the functionality of the logging driver if they want to make any changes to it.

+1 on generalizing the function call, I can rename it.